### PR TITLE
Expose the fully resolved image tags generated by the Docker plugin.

### PR DESCRIPTION
**Pull request by ColinArenz**
These tags will be needed by the `dockerPush` command and also for scripted deployment.  Reconstructing the tags given only a project name would be a bit messy and error-prone outside of SBT.

### CODE REVIEW

File: src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala [hunk 1]
```diff
@@ -534,6 +553,20 @@ $DOCKERFILE_SIGIL
...
+    val imageTag = imageNameWithLabel(dependencyImageDir.value, dependencyImageName.value)
+    val logger = Keys.streams.value.log
+    logger.info(imageTag)
+    imageTag
+  }
+
+  lazy val mainImageTagDef: Def.Initialize[Task[String]] = Def.task {
+    val imageTag = imageNameWithLabel(mainImageDir.value, mainImageName.value)
+    val logger = Keys.streams.value.log
+    logger.info(imageTag)
```
Comment (jkinkead) [inline src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala]: If you use the `show` command in the sbt console, you don't need this log. It literally does exactly this.

File: src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala [hunk 2]
```diff
@@ -138,6 +138,18 @@ object DockerBuildPlugin extends AutoPlugin {
     val dockerKill: TaskKey[Unit] = taskKey[Unit](
       "Kills any currently-running docker container for this project."
     )
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // The following keys expose information about the docker file being generated.
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+
+    val dockerDependencyImageTag: TaskKey[String] = taskKey[String](
+      "Generates and prints the tag of the dependency image that `dockerBuild` will produce."
```
Comment (ColinArenz) [inline src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala]: I'm not actually sure... I assumed I had to push both.  If not, I'll remove it.

File: src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala [hunk 3]
```diff
@@ -171,6 +183,15 @@ object DockerBuildPlugin extends AutoPlugin {
   // Common settings / tasks / utilities used across tasks.
   //////////////////////////////////////////////////////////////////////////////////////////////////
 
+  def hashDirectory(imageDir: File): String = {
+    val allFiles = PathFinder(imageDir).***.filter(_.isFile).get
+    Utilities.hashFiles(allFiles, imageDir)
+  }
+
+  def imageNameWithLabel(imageDir: File, imageName: String): String = {
```
Comment (ColinArenz) [inline src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala]: Removed the directory hashing so it has to be called explicitly each time. This reduces the function to a simple string concatenation, but I think we should keep it for expressiveness.

**Comment by ColinArenz** [comment]
From offline: There are a few other ways to solve this problem that don't involve this plugin.  I'm going to retract this change, at least for now.

**Pull request closed without merging**